python/ngen_conf/src/ngen/config/init_config/noahowp.py

```diff
@@ -0,0 +1,248 @@
+                "description": "rain-snow temperature threshold [degrees C]"
+            },
+        }
+
+
+class ModelOptions(core.Base):
+    # option bounds translated from
+    # https://github.com/NOAA-OWP/noah-owp-modular/blob/30d0f53e8c14acc4ce74018e06ff7c9410ecc13c/src/OptionsType.f90#L11
+    precip_phase_option: PrecipPhaseOption
+    snow_albedo_option: SnowAlbedoOption
+    dynamic_veg_option: DynamicVegOption
+    runoff_option: RunoffOption
+    drainage_option: DrainageOption
+    frozen_soil_option: FrozenSoilOption
+    dynamic_vic_option: DynamicVicOption
+    radiative_transfer_option: RadiativeTransferOption
+    sfc_drag_coeff_option: SfcDragCoeffOption
+    canopy_stom_resist_option: CanopyStomResistOption
+    crop_model_option: CropModelOption
+    snowsoil_temp_time_option: SnowsoilTempTimeOption
+    soil_temp_boundary_option: SoilTempBoundaryOption
+    supercooled_water_option: SupercooledWaterOption
+    stomatal_resistance_option: StomatalResistanceOption
+    evap_srfc_resistance_option: EvapSrfcResistanceOption
+    subsurface_option: SubsurfaceOption
+
+    class Config(core.Base.Config):
+        # TODO: consider serializing all enum's using their value unless otherwise specified
+        field_serializers = {
+            "precip_phase_option": serialize_enum_value,
+            "snow_albedo_option": serialize_enum_value,
+            "dynamic_veg_option": serialize_enum_value,
+            "runoff_option": serialize_enum_value,
+            "drainage_option": serialize_enum_value,
+            "frozen_soil_option": serialize_enum_value,
+            "dynamic_vic_option": serialize_enum_value,
+            "radiative_transfer_option": serialize_enum_value,
+            "sfc_drag_coeff_option": serialize_enum_value,
+            "canopy_stom_resist_option": serialize_enum_value,
+            "crop_model_option": serialize_enum_value,
+            "snowsoil_temp_time_option": serialize_enum_value,
+            "soil_temp_boundary_option": serialize_enum_value,
+            "supercooled_water_option": serialize_enum_value,
+            "stomatal_resistance_option": serialize_enum_value,
+            "evap_srfc_resistance_option": serialize_enum_value,
+            "subsurface_option": serialize_enum_value,
+        }
+
+
+class LandSurfaceType(int, Enum):
+    """
+    land surface type, 1: soil, 2: lake
+    source: https://github.com/NOAA-OWP/noah-owp-modular/blob/30d0f53e8c14acc4ce74018e06ff7c9410ecc13c/src/NamelistRead.f90#L37
+    """
+
+    soil = 1
+    lake = 2
+
+
+class Structure(core.Base):
+    isltyp: int  #    = 1
+    nsoil: int  #     = 4
+    nsnow: int  #     = 3
+    nveg: int  #      = 20
+    vegtyp: int  #    = 1
+    # crop type (SET TO 0, no crops currently supported)
+    # source: https://github.com/NOAA-OWP/noah-owp-modular/blob/30d0f53e8c14acc4ce74018e06ff7c9410ecc13c/src/NamelistRead.f90#L36
+    croptype: int = 0
+    sfctyp: LandSurfaceType  #    = 1
```

Comment:
@SnowHydrology, the [source](https://github.com/NOAA-OWP/noah-owp-modular/blob/30d0f53e8c14acc4ce74018e06ff7c9410ecc13c/src/NamelistRead.f90#L36
) of `noahowp` listed the meaning of the two `sfctyp` variants (soil or lake), but the others did not.

```fortran
  integer            :: croptype           ! crop type (SET TO 0, no crops currently supported)
```

Can point me in the direction of the other fields variant names? It would be great to include them too.

Reply:
Ah, that makes sense. The list of `vegtyp` options depends on `veg_class_name`.